packages/application-components/src/components/tab-header/tab-header.tsx

```diff
@@ -0,0 +1,128 @@
+import { Link, matchPath, useLocation } from 'react-router-dom';
+import startCase from 'lodash.startcase';
+import { FormattedMessage, type MessageDescriptor } from 'react-intl';
+import type { ReactNode, ElementType, MouseEvent, KeyboardEvent } from 'react';
+import type { LocationDescriptor } from 'history';
+import { getTabHeaderStyles, getLinkWrapperStyles } from './tab.styles';
+import { warning } from '@commercetools-uikit/utils';
+
+const pathWithoutSearch = (path: TTabHeaderProps['to']) =>
+  typeof path === 'string' ? path.split('?')[0] : path.pathname;
+
+const noop = () => {};
+
+const warnIfMissingContent = (props: TTabHeaderProps) => {
+  const hasContent = Boolean(props.intlMessage) || Boolean(props.children);
+
+  warning(
+    hasContent,
+    'TabHeader: one of either `children` or `intlMessage` is required but their values are `undefined`'
+  );
+};
+
+type TLinkWrapperProps = {
+  to: TTabHeaderProps['to'] | null;
+  children: ReactNode;
+};
+
+/* This wrapper component provides disabled mode for `TabHeader`.
+If `to` prop is `null` the wrapping `<a>` element (used instead of `<Link>`) has no `href` attribute.
+Therefore, it is not clickable and falls out of the tabbing sequence.  */
+const LinkWrapper = (props: TLinkWrapperProps) => {
+  const Component = (props.to ? Link : 'a') as ElementType;
+  return <Component {...props}>{props.children}</Component>;
+};
+LinkWrapper.displayName = 'LinkWrapper';
+
+export type TTabHeaderProps = {
+  /**
+   * Path that the TabHeader directs to when not disabled.
+   */
+  to: string | LocationDescriptor;
+  /**
+   * Any React node passed to TabHeader.
+   * * <br />
+   * Required if `intlMessage` is not provided.
+   */
+  children?: ReactNode;
```

Comment:
On second thoughts, I think we don't even need `children` as I don't see a use case where you need to render anything other than a string.
I would suggest to add instead a new prop `label?: string`.

So you essentially pass a string label to either `label` or as `intlMessage` (when using translations). We don't allow any other React element.

The `name` prop then can also be removed, as we can then simply use the string value of label (or intlMessage).